### PR TITLE
@W-13985181: [MSDK Android] Remove Unneeded Show Developer Support Activity

### PR DESCRIPTION
**Pull request by JohnsonEricAtSalesforce**
🚧 _Under Construction_ 🧱

This builds on the [discussion](https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2446#discussion_r1295377518) in #2446 to try using the new notification's content intent with a broadcast intent, which removes the need for a counterintuitive trigger activity that shows the developer support dialog.

The only consider against is that Material design introduced an inspector warning when using broadcast intents as the content intent for notifications.  The documentation simply states that a user expects to see a new activity detailing the notification by starting an activity from a notification action instead.

MSDK's use case really is different from Material's recommendation, though.  MSDK wants to display a dialog from the visible activity.  Thinking of that difference, perhaps suppressing the inspector warning is a valid approach.

Let's take a look and see which approach seems preferable.  Thanks!

### CODE REVIEW

**Comment by JohnsonEricAtSalesforce** [comment]
By the by, I targeted the branch for #2446 for now so we can see the diff as it will look after #2446 is merged.  I'll edit the branch as we merge pull requests following the 11.0.1 patch.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java [hunk 1]
```diff
@@ -907,12 +915,6 @@ private void updateDeveloperSupportForActivityLifecycle(
             @NonNull Activity lifecycleActivity
     ) {
 
-        // Show the developer support dialog, if applicable.
-        if (lifecycleActivity.isDestroyed() && lifecycleActivity instanceof ShowDeveloperSupportActivity) {
```
Comment (JohnsonEricAtSalesforce) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java]: Here's the elimination of responding to the trigger activity's finish.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java [hunk 2]
```diff
@@ -987,17 +989,27 @@ public void onActivitySaveInstanceState(@NonNull Activity activity, @NonNull Bun
...
-                    );
-                }
+                /* Intentionally Blank */
             }
         };
         application.registerActivityLifecycleCallbacks(activityLifecycleCallbacks);
         salesforceSDKManager.activityLifecycleCallbacksForDeveloperSupport = activityLifecycleCallbacks;
+
+        // Register an Android broadcast intent receiver to respond to the show developer support notification.
+        final BroadcastReceiver showDeveloperSupportBroadcastIntentReceiver = new BroadcastReceiver() {
```
Comment (JohnsonEricAtSalesforce) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java:1025]: Here's the new broadcast intent receiver.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/developer/support/notifications/local/ShowDeveloperSupportNotifier.kt [hunk 3]
```diff
@@ -93,8 +97,9 @@ internal class ShowDeveloperSupportNotifier {
         /**
          * Displays the Salesforce Mobile SDK "Show Developer Support"
          * notification.
-         * * @param activity The Android activity
+         * @param activity The Android activity
          */
+        @SuppressLint("LaunchActivityFromNotification")
```
Comment (JohnsonEricAtSalesforce) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/developer/support/notifications/local/ShowDeveloperSupportNotifier.kt:109]: This suppress is what the trigger activity aided in avoiding.  Thinking on it a bit longer though, our difference from the use case described by the Material design docs makes this suppress logical.  This notification doesn't have a dedicated activity to display as Material suggests - it's triggering an action in the visible activity.

**Review by JohnsonEricAtSalesforce — COMMENTED** [review]
Here's some self-review notes to introduce the changes.

**Comment by JohnsonEricAtSalesforce** [comment]
This will still need its target branch set after #2446 is merged to upstream `dev`, but this is reviewable so we can determine if it will be useful or if we prefer to close out this idea.

**Review by brandonpage — APPROVED** [review]
🎉

**Pull request closed without merging**